src/backend/distributed/commands/begin.c

```diff
@@ -0,0 +1,49 @@
+/*-------------------------------------------------------------------------
+ *
+ * begin.c
+ *    Processing of the BEGIN command.
+ *
+ * Copyright (c) Citus Data, Inc.
+ *
+ *-------------------------------------------------------------------------
+ */
+
+#include "postgres.h"
+#include "c.h"
+
+#include "distributed/commands.h"
+#include "distributed/listutils.h"
+#include "distributed/transaction_management.h"
+#include "nodes/parsenodes.h"
+
+
+/*
+ * CitusProcessBegin handles TransactionStmt that has already been qualified
+ * as a BEGIN statement.
+ */
+void
+CitusProcessBegin(TransactionStmt *transactionStmt)
+{
+	DefElem *item = NULL;
+
+	/*
+	 * This loop is similar to the one in standard_ProcessUtility.
+	 *
+	 * While BEGIN can be quite frequent it will rarely have options set.
+	 */
+	foreach_ptr(item, transactionStmt->options)
+	{
+		A_Const *constant = (A_Const *) item->arg;
+
+		if (strcmp(item->defname, "transaction_read_only") == 0)
```

Comment:
btw, any READ ONLY hits assertion failure:

```SQL
 BEGIN TRANSACTION ISOLATION LEVEL SERIALIZABLE DEFERRABLE READ ONLY;
server closed the connection unexpectedly
	This probably means the server terminated abnormally
	before or while processing the request.
The connection to the server was lost. Attempting reset: Failed.
The connection to the server was lost. Attempting reset: Failed.
Time: 12.183 ms
 @:-!> 
```

bt:
```
* thread #1, queue = 'com.apple.main-thread', stop reason = signal SIGABRT
  * frame #0: 0x00007ff804f0030e libsystem_kernel.dylib`__pthread_kill + 10
    frame #1: 0x00007ff804f37f7b libsystem_pthread.dylib`pthread_kill + 263
    frame #2: 0x00007ff804e81ca5 libsystem_c.dylib`abort + 123
    frame #3: 0x000000010c89582f postgres`ExceptionalCondition(conditionName=<unavailable>, errorType=<unavailable>, fileName=<unavailable>, lineNumber=<unavailable>) at assert.c:69:2
    frame #4: 0x000000010d2a96b5 citus.so`castNodeImpl(type=T_Boolean, ptr=<unavailable>) at nodes.h:609:2
    frame #5: 0x000000010d2a95df citus.so`CitusProcessBegin(transactionStmt=0x00007f8cc300ac00) at begin.c:0:2
    frame #6: 0x000000010d2c912f citus.so`multi_ProcessUtility(pstmt=0x00007f8cc300af70, queryString="BEGIN TRANSACTION READ ONLY;", readOnlyTree=<unavailable>, context=PROCESS_UTILITY_TOPLEVEL, params=0x0000000000000000, queryEnv=<unavailable>, dest=0x00007f8cc300b060, completionTag=0x00007ff7b3ac3ef0) at utility_hook.c:185:4
    frame #7: 0x000000010cfc5515 pg_stat_statements.so`pgss_ProcessUtility(pstmt=0x00007f8cc300af70, queryString=<unavailable>, readOnlyTree=<unavailable>, context=PROCESS_UTILITY_TOPLEVEL, params=0x0000000000000000, queryEnv=0x0000000000000000, dest=0x00007f8cc300b060, qc=0x00007ff7b3ac3ef0) at pg_stat_statements.c:1139:5
```

Reply:
oops, that's a pg15 vs. pg14/pg13 difference, will fix it